pkg/controller/node/nodecontroller.go

```diff
@@ -573,6 +668,42 @@ func (nc *NodeController) monitorNodeStatus() error {
 	return nil
 }
 
+// evictIntolerablePods checks for given node if there's any pod that
```

Comment:
BTW it's OK if you don't handle all of those scenarios yet, but at least add a comment explaining which ones you do and which ones you don't. (And some information about under what circumstances this function gets invoked.)